pyp2spec/pyp2conf.py

```diff
@@ -43,26 +45,46 @@ def pypi_name(self):
 
         return self.normalize(self.package_data["info"]["name"])
 
+    @property
+    def version(self):
+        """Set either to the custom package version provided by the user
+        or to the latest one from package metadata.
+        """
+
+        return self._version or self._get_version_from_package_metadata()
```

Comment:
This property seems to query PyPI every time when read (unless `self._version` is set).

That seems wasteful and might create a race condition when  new version is uploaded to PyPI mid-run.

What if the property did something like this:

```suggestion
        self._version = self._version or self._get_version_from_package_metadata()
        return self._version
```

Obviously, the docstring would need updating.

Reply:
shoot, good catch

Reply:
remedied in: b635a56819600c0e1d6ef5481154736f26aaf7f5